packages/hash/api/src/model/entity.model.ts

```diff
@@ -614,25 +614,30 @@ class __Entity {
         })),
       );
 
-      await Promise.all(
-        entities.map(({ link, entity }) => {
-          if (link) {
-            const parentEntity = entities[link.parentIndex];
-            if (!parentEntity) {
-              throw new ApolloError("Could not find parent entity");
-            }
-            // links are created as an outgoing link from the parent entity to the children.
-            return parentEntity.entity.createOutgoingLink(client, {
-              createdByAccountId: user.accountId,
-              destination: entity,
-              stringifiedPath: link.meta.path,
-              index: link.meta.index ?? undefined,
-            });
-          } else {
-            return null;
+      /**
+       * Currently the outgoing links of an entity cannot be reliably created in parallel,
+       * because when indexed links are created the affected indexed links which have the same path
+       * and source entity have to be udpated. Running these operations in parallel can result in the
+       * update methods being run out of order, resulting in incorrect `index` values for the affected links.
+       *
+       * @todo: find a way to run these in parallel - potentially by implementing a `createLinks` adapter method,
+       * or by locking/releasing the relevant rows of affected outgoing links in the `createLink` adapter method.
```

Comment:
Have created a task for this [here](https://app.asana.com/0/1200211978612931/1201997896999031/f)